File: Swift/largest-time-for-given-digits.swift
class Solution {
    func largestTimeFromDigits(_ A: [Int]) -> String {
        var res = -1

        for i in 0..<4 {
            for j in 0..<4 {
                if j != i {
                    for k in 0..<4 {
                        if k != i && k != j {
                            let l = 6 - i - j - k
                            let hour = 10 * A[i] + A[j]
                            let min = 10 * A[k] + A[l]
                            if hour < 24 && min < 60 {
                                res = max(res, hour * 60 + min)
                            }
                        }
                    }
                }
            }
        }

        return  res >= 0 ? String(format: "%02d:%02d", res / 60, res % 60) : ""
    }
}

File: Python/largest-time-for-given-digits.py
import itertools

class Solution:
    def largestTimeFromDigits(self, A):
        """
        :type A: List[int]
        :rtype: str
        """
        res = ""

        for i in range(len(A)):
            A[i] *= -1
        A.sort()

        for h1, h2, m1, m2 in itertools.permutations(A):
            hours = -(10 * h1 + h2)
            mins = -(10 * m1 + m2)
            if 0 <= hours < 24 and 0 <= mins < 60:
                res = "{:02}:{:02}".format(hours, mins)
                break

        return res
